### PR TITLE
fix equals and hashcode issue

### PR DESCRIPTION
**Pull request by seagle-yuan**
fix equals and hashcode issue

com.baidu.hugegraph.auth.RolePermission.java
com.baidu.hugegraph.backend.cache.RamCache.java
com.baidu.hugegraph.backend.serializer.BinaryBackendEntry.java
com.baidu.hugegraph.backend.serializer.TextBackendEntry.java
com.baidu.hugegraph.structure.HugeEdgeProperty.java
com.baidu.hugegraph.traversal.optimize.HugeCountStep.java
com.baidu.hugegraph.traversal.optimize.HugeGraphStep.java
com.baidu.hugegraph.traversal.optimize.HugeVertexStep.java

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/cache/RamCache.java [hunk 1]
```diff
@@ -252,6 +252,11 @@ public boolean equals(Object obj) {
             LinkNode<K, V> other = (LinkNode<K, V>) obj;
             return this.key().equals(other.key());
         }
+
+        public int hashCode() {
+            return this.key().hashCode();
+        }
+
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/cache/RamCache.java:259]: remove useless blank line

**Review by javeme — APPROVED** [review]

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BinaryBackendEntry.java [hunk 2]
```diff
@@ -197,6 +197,11 @@ public boolean equals(Object obj) {
         return this.columns.containsAll(other.columns);
     }
 
+    public int hashCode() {
+        return this.id().hashCode() ^
+               this.columns.size();
```
Comment (imbajin) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BinaryBackendEntry.java:202]: why we separate them to 2 lines?

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/TextBackendEntry.java [hunk 3]
```diff
@@ -371,4 +372,13 @@ public boolean equals(Object obj) {
         }
         return true;
     }
+
+    // ConcurrentSkipListMap override equals();
+    // ConcurrentSkipListMap extend AbstractMap and AbstractMap implement
+    // hashCode()
```
Comment (imbajin) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/TextBackendEntry.java:378]: useful for future? or use `/**/` for it

Comment (seagle-yuan) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/TextBackendEntry.java:378]: it's unusefun,delete this comments;

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BinaryBackendEntry.java (hunk 2 above)
Comment (seagle-yuan) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BinaryBackendEntry.java:202]: merge into one line

**Review by imbajin — APPROVED** [review]

**Review by javeme — APPROVED** [review]

**Pull request merged**